src/Toolkit/Toolkit.WPF/UI/Controls/PopupViewer/TextPopupElementView.cs

```diff
@@ -0,0 +1,189 @@
+        /// Gets or sets the TextPopupElement.
+        /// </summary>
+        public TextPopupElement? Element
+        {
+            get { return GetValue(ElementProperty) as TextPopupElement; }
+            set { SetValue(ElementProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="Element"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty ElementProperty =
+            DependencyProperty.Register(nameof(Element), typeof(TextPopupElement), typeof(TextPopupElementView), new PropertyMetadata(null, (s, e) => ((TextPopupElementView)s).OnElementPropertyChanged()));
+
+        private void OnElementPropertyChanged()
+        {
+            // TODO: Convert to pretty html
+            // Full list of supported tags and attributes here: https://doc.arcgis.com/en/arcgis-online/reference/supported-html.htm
+            return;
+            if (!string.IsNullOrEmpty(Element?.Text))
+            {
+                HtmlTokenParser parser = new HtmlTokenParser(Element.Text);
+                FlowDocument doc = new FlowDocument();
+                bool isBold = false;
+                bool isItalic = false;
+                bool isHyperLink = false;
+                while (parser.NextToken(out HtmlToken? t))
+                {
+                    var token = t.Value;
+                    if (token.Type == TokenType.Begin && token.Name == "p")
+                        doc.Blocks.Add(new Paragraph(new Span()));
+                    else if (doc.Blocks.Count == 0)
+                        doc.Blocks.Add(new Paragraph(new Span()));
+
+                    if (token.Name == "b" || token.Name == "strong")
+                        isBold = token.Type == TokenType.Begin;
+                    else if (token.Name == "i" || token.Name == "em")
+                        isItalic = token.Type == TokenType.Begin;
+                    else if (token.Name == "a")
+                        isHyperLink = token.Type == TokenType.Begin;
+
+                    if (token.Name == "br" && token.Type !=  TokenType.End)
+                    {
+                        ((Span)((Paragraph)doc.Blocks.Last()).Inlines.Last()).Inlines.Add(new LineBreak());
+                    }
+                    if (token.Type == TokenType.None)
+                    {
+                        ((Span)((Paragraph)doc.Blocks.Last()).Inlines.Last()).Inlines.Add(new TextBlock()
+                        {
+                            Text = token.Name,
+                            FontWeight = isBold ? FontWeights.Bold : FontWeights.Normal,
+                            FontStyle = isItalic ? FontStyles.Italic : FontStyles.Normal,
+                        });
+                    }
+                }
+                if (GetTemplateChild("TextArea") is RichTextBox rtb)
+                {
+                    rtb.Document = doc;
+                }
+            }
+        }
+
+        private class HtmlTokenParser
```

Comment:
This class is just begging to be unit-tested! 😄

Reply:
It's not used yet - it's a todo for you ;-)